casper/src/main/scala/coop/rchain/casper/util/comm/CasperPacketHandler.scala

```diff
@@ -0,0 +1,496 @@
+
+object CasperPacketHandler extends CasperPacketHandlerInstances {
+  private implicit val logSource: LogSource = LogSource(this.getClass)
+
+  def apply[F[_]](implicit ev: CasperPacketHandler[F]): CasperPacketHandler[F] = ev
+
+  def of[
+      F[_]: LastApprovedBlock: Metrics: Timer: BlockStore: ConnectionsCell: NodeDiscovery: TransportLayer: ErrorHandler: RPConfAsk: SafetyOracle: Capture: Sync: Time: Log: MultiParentCasperRef](
+      conf: CasperConf,
+      delay: FiniteDuration,
+      runtimeManager: RuntimeManager,
+      toTask: F[_] => Task[_])(implicit scheduler: Scheduler): F[CasperPacketHandler[F]] =
+    if (conf.approveGenesis) {
+      for {
+        genesis <- Genesis.fromInputFiles[F](conf.bondsFile,
+                                             conf.numValidators,
+                                             conf.genesisPath,
+                                             conf.walletsFile,
+                                             runtimeManager,
+                                             conf.shardId,
+                                             conf.deployTimestamp)
+        validatorId <- ValidatorIdentity.fromConfig[F](conf)
+        bap         = new BlockApproverProtocol(validatorId.get, genesis, conf.requiredSigs)
+        gv          <- Ref.of[F, CasperPacketHandlerInternal[F]](new GenesisValidatorHandler(bap))
+      } yield new CasperPacketHandlerImpl[F](gv)
+    } else if (conf.createGenesis) {
+      for {
+        genesis <- Genesis.fromInputFiles[F](conf.bondsFile,
+                                             conf.numValidators,
+                                             conf.genesisPath,
+                                             conf.walletsFile,
+                                             runtimeManager,
+                                             conf.shardId,
+                                             conf.deployTimestamp)
+        validatorId      <- ValidatorIdentity.fromConfig[F](conf)
+        bondedValidators <- Genesis.getBondedValidators[F](conf.bondsFile)
+        abp <- ApproveBlockProtocol
+                .of[F](genesis,
+                       bondedValidators,
+                       conf.requiredSigs,
+                       conf.approveGenesisDuration,
+                       conf.approveGenesisInterval)
+                .map(protocol => {
+                  toTask(protocol.run()).forkAndForget.runAsync
+                  protocol
+                })
+        standalone <- Ref.of[F, CasperPacketHandlerInternal[F]](new StandaloneCasperHandler[F](abp))
+        _ <- Sync[F].delay {
+              val _ = toTask(
+                StandaloneCasperHandler
+                  .approveBlockInterval(conf.approveGenesisInterval,
+                                        conf.shardId,
+                                        runtimeManager,
+                                        validatorId,
+                                        standalone)).forkAndForget.runAsync
```

Comment:
Is the val _ needed?

Reply:
Not really but sometimes compiler complains about statements that return values which are not assigned to anything (in this case they aren't really assigned to anything either but it's enough).